python/lsst/ci/git.py

```diff
@@ -26,7 +26,8 @@ def __call__(self, *args, **kwargs):
 
         return_status = kwargs.get("return_status", False)
 
-        cmd = ('git',) + args
+        # force all cli args into strings
+        cmd = ['git'] + [str(x) for x in args]
```

Comment:
Now that the git command is sufficiently complex, have you considered using something like GitPython over Popen directly? There is quite a bit of error handling and cleanup that GitPython provides.